src/internal/connector/exchange/data_collections.go

```diff
@@ -182,8 +182,11 @@ func DataCollections(
 		categories  = map[path.CategoryType]struct{}{}
 	)
 
-	// TODO: Add hidden cli flag to disable this feature
-	graph.InitializeConcurrencyLimiter(ctrlOpts.Parallelism.ItemFetch)
+	// Turn on concurrency limiter middleware for exchange backups
+	// unless explicitly disabled through the --disable-concurrency-limiter cli flag
```

Comment:
Out of curiosity: under what conditions to we expect to want to run without the concurrency limiter in place?

Reply:
We have done limited perf testing with the concurrency limiter. In addition, our understanding of exchange concurrency limits is also a bit fuzzy. This flag acts as a killswitch in case we start to see perf regressions in prod. 

I see this as a temporary flag. Ideally we would never have to use it. Also, once we have a better way to configure concurrency limiter, we can eliminate this flag.